haskell/src/Stablecoin/Client/Impl.hs

```diff
@@ -327,12 +316,55 @@ getMintingAllowance contract minter = do
   let allowanceMaybe = M.lookup minterAddr mintingAllowances
   pure $ fromMaybe 0 allowanceMaybe
 
--- | Get the token metadata of the contract
 getTokenMetadata :: "contract" :! AddressOrAlias -> MorleyClientM FA2.TokenMetadata
 getTokenMetadata contract = do
-  mdRegistry <- sTokenMetadataRegistry <$> getStorage contract
-  bigMapId <- mrsTokenMetadata <$> getRegistryStorage mdRegistry
-  readBigMapValue bigMapId FA2.theTokenId
+  (metadata, storageView) <- getContractMetadata contract
+  case TZ.getViews metadata of
+    Left err -> throwM $ SCEMetadataError ("Views was not found in metadata:" <> show err)
+    Right views -> case MD.findView @(ToT Storage) views "GetTokenMetadata" of
+      Nothing -> throwM $ SCEMetadataError "'GetTokenMetadata' view was not found in metadata"
+      Just getTMD -> do
+        let storageWithEmptyBm :: Storage =
+              -- In the below code we convert the `StorageView` to `Storage` by
+              -- replacing all bigmaps with empty counterparts, and this should be fine here since
+              -- the token metadata view does not access any of these big maps.
+              storageView
+                { sLedger = mempty
+                , sMetadata = mempty
+                , sOperators = mempty
+                , sPermits = mempty
+                }
+        case MD.interpretView @(Natural, Map MText ByteString)
+            dummyContractEnv getTMD (MD.ViewParam (0 :: Natural)) storageWithEmptyBm of
+          Right (_, md) -> pure md
+          Left err -> throwM $ SCEMetadataError ("Error while interpreting the contract:" <> show err)
```

Comment:
I forgot to write this, but `throwMdErr` could be used in this function as well to the same effect.

Reply:
@sras my comment was not very clear, I didn't mean `MD.interpretView` specifically, but `getTokenMetadata` in general.

It can also be used with `TZ.getViews` (and with `maybeToRight` perhaps even with `MD.findView`).